DFC.Digital/DFC.Digital.Service.CourseSearch/Converters/MessageConverter.cs

```diff
@@ -68,15 +68,22 @@ internal static IEnumerable<Course> ConvertToSearchCourse(this CourseListOutput
             return result ?? Enumerable.Empty<Course>();
         }
 
-        internal static CourseDetails ConvertToCourseDetails(this CourseDetailOutput apiResult)
+        internal static CourseDetails ConvertToCourseDetails(this CourseDetailOutput apiResult, string oppurtunityId, string courseId)
         {
-            var apiCourseDetail = apiResult.CourseDetails?.FirstOrDefault();
+            var apiCourseDetail = apiResult.CourseDetails?.Where(co => co.Course.CourseID == courseId)?.FirstOrDefault();
 
             if (apiCourseDetail == null)
             {
                 return null;
             }
 
+            var activeOppurtunity = apiCourseDetail.Opportunity.FirstOrDefault();
```

Comment:
Provide reason comment for using "FirstOrDefault"

Reply:
When citizen comes from course search results page to there wont be any opportunity Id and that's when we are choosing the first opportunity in the list